ios/react-native-healthkit/RCTHealthKit+Characteristics.m

```
@@ -37,18 +37,19 @@ - (void)_getDefaultSource:(RCTPromiseResolveBlock)resolve
 
 #pragma mark - Workouts
 
-- (void)_addWorkout:(NSDate*)startDate
+- (void)_addWorkout:(NSString*)activityType
+          startDate:(NSDate*)startDate
             endDate:(NSDate*)endDate
-            calories:(float)calories
+           calories:(float)calories
    distanceInMeters:(float)distance
-            metadata:(NSDictionary*)metadata
+           metadata:(NSDictionary*)metadata
             resolve:(RCTPromiseResolveBlock)resolve
-            reject:(RCTPromiseRejectBlock)reject{
+             reject:(RCTPromiseRejectBlock)reject{
     HKUnit *caloriesUnit = [HKUnit kilocalorieUnit];
     HKQuantity *caloriesQuantity = [HKQuantity quantityWithUnit:caloriesUnit doubleValue:calories];
     HKQuantity *distanceQuantity = distance != -1 ? [HKQuantity quantityWithUnit:HKUnit.meterUnit doubleValue:distance] : nil;
-
-    HKWorkout *workout = [HKWorkout workoutWithActivityType:HKWorkoutActivityTypeOther
+    
+    HKWorkout *workout = [HKWorkout workoutWithActivityType:[RCTHealthKitDataModels.workoutsDict[activityType] intValue]
```

I belive a `#import "RCTHealthKitDataModels.h"` is missing on this file, while building I received a `use of undeclared identifier 'RCTHealthKitDataModels'`.

Also I would check if `RCTHealthKitDataModels.workoutsDict[activityType]` is `nil` before passing to `HKWorkout workoutWithActivityType` since the `activityType` will come from the JS side as a string is very possible we end up masking errors due to typos